### PR TITLE
Mapgen

### LINKED ISSUE #5: test mapgeneration

### PR DESCRIPTION
**Pull request by jesings**
Ok now actually

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,6 +7,7 @@ edition = "2018"
 
 [dependencies]
 rand = "0.8"
+rand_xoshiro = "0.6"
```
Comment (anotherLostKitten) [inline Cargo.toml:10]: this sounds like some weeb shit

File: src/map/mapgen.rs [hunk 2]
```diff
@@ -1,25 +1,129 @@
...
+                    let gridtiles = scratchvec.drain(..).map(|i| tileset[i as usize].clone()).collect();
+
+                    return Self {
+                        cols: width,
+                        rows: height,
+                        tiles: gridtiles
+                    };
+                }
+            }
+            println!("Grid generation failed! Retrying to generate grid!");
```
Comment (anotherLostKitten) [inline src/map/mapgen.rs]: this should happen silently maybe?

File: src/server_gameloop.rs [hunk 3]
```diff
@@ -60,18 +76,26 @@ pub fn gameloop() {
...
+
     let mut playarrs = vec!();
     for i in 0..streams.len() {
         playarrs.push(Player::test_player(i));
     }
 
     let gd = Arc::new(Gamedata {
         players: playarrs.drain(..).map(|x| Arc::new(Mutex::new(x))).collect(),
-        grid: Grid::gen_blank_grid(640, 480),
+        grid: Grid::gen_cell_auto(640, 480, mapseed),
```
Comment (anotherLostKitten) [inline src/server_gameloop.rs]: constants

File: src/server_gameloop.rs [hunk 4]
```diff
@@ -60,18 +76,26 @@ pub fn gameloop() {
...
     });
+
+
     let handles = servnet::launch_server_workers(streams, gd.clone(), mpsc_tx, &mut spmc, livelisteners.clone());
 
+    spmc.broadcast(Arc::new(PktPayload::Gamedata(GDTuple {0: gd.players.iter().map(|x| (*x.lock().unwrap()).clone()).collect(), 1: mapseed, 2: 343})));
```
Comment (anotherLostKitten) [inline src/server_gameloop.rs:97]: why is it 343

**Review by anotherLostKitten — APPROVED** [review]
this is

File: Cargo.toml (hunk 1 above)
Comment (jesings) [reply Cargo.toml:10]: It's not it's from italy
https://prng.di.unimi.it/

File: src/map/mapgen.rs (hunk 2 above)
Comment (jesings) [reply src/map/mapgen.rs]: Maybe but it's a really exceptional case and it's kind of useful to tell if things might be taking somewhat longer?
I wasn't sure whether to keep it in or leave it out.

File: Cargo.toml (hunk 1 above)
Comment (anotherLostKitten) [reply Cargo.toml:10]: oh ok 👍

File: src/map/mapgen.rs (hunk 2 above)
Comment (anotherLostKitten) [reply src/map/mapgen.rs]: yes i wasn't sure how often this happened so maybe it's ok but should it at least be printerr or something then?

**Pull request merged**